File: server/router/disk.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitDiskRouter(Router *gin.RouterGroup) {
	DiskRouter := Router.Group("disk").
		Use(middleware.JWTAuth()).
		Use(middleware.OperationRecord()).
		Use(middleware.CasbinHandler())
	{
		DiskRouter.POST("getDiskList", v1.GetDiskList)  // 分页获取硬盘列表
		DiskRouter.POST("createDisk", v1.CreateDisk)    //创建硬盘
		DiskRouter.POST("updateDisk", v1.UpdateDisk)    //更新硬盘信息
		DiskRouter.POST("getDiskById", v1.GetDiskById)  //获取盘通过id
		DiskRouter.POST("deleteDisk", v1.DeleteDisk)    //删除盘
	}
}

File: web/src/api/disk.js
import service from '@/utils/request'


// 获取硬盘信息
export const getDiskList = (data) => {
  return service({
      url: "/disk/getDiskList",
      method: 'post',
      data: data
  })
}

//新增硬盘
export const createDisk = (data) => {
  return service({
      url: "/disk/createDisk",
      method: 'post',
      data: data
  })
}

//修改硬盘
export const updateDisk = (data) => {
  return service({
      url: "/disk/updateDisk",
      method: 'post',
      data: data
  })
}

//  获取盘通过id
export const getDiskById = (data) => {
  return service({
      url: "/disk/getDiskById",
      method: 'post',
      data: data
  })
}

//删除硬盘
export const deleteDisk = (data) => {
  return service({
      url: "/disk/deleteDisk",
      method: 'post',
      data: data
  })
}
